bundle/edu.gemini.phase2.skeleton.servlet/src/main/scala/edu/gemini/phase2/template/factory/impl/TemplateFactoryImpl.scala

```diff
@@ -83,7 +85,9 @@ case class TemplateFactoryImpl(db: TemplateDb) extends TemplateFactory {
 
       // GNIRS
       case b: SpGnirsBlueprintImaging => Right(GnirsImaging(b))
-      case b: SpGnirsBlueprintSpectroscopy => Right(GnirsSpectroscopy(b, sampleTarget))
+      case b: SpGnirsBlueprintSpectroscopy =>
+        if (b.fpu.isIfu) Right(GnirsSpectroscopyIfu(b, sampleTarget))
+        else Right(GnirsSpectroscopy(b, sampleTarget))
```

Comment:
Structurally the blueprint is the same for IFU or not, but the expansion logic is different enough that I split it in up, hence the `if` here.